feat: Maestro Bitcoin indexer

mhrynenko opened the pull request.

mhrynenko committed:
wip: introduce base for Maestro Bitcoin indexer, setup fabric pattern
for indexer spawner

mhrynenko committed:
wip: create direct integration with Maestro api, restore btc_indexer
Dockerfile,

mhrynenko committed:
chore: rename method for clarity

mhrynenko committed:
feat: add config env source, return Proper error from request maker,
make some model fields optional

mhrynenko committed:
refactor: avoid using titan client, move into existing general
implementation for btc indexer client, split direct dependencies in
tests managers, extend indexer_client trait methods

kevandee commented on tests/src/user_wallet.rs:
```diff
@@ -70,8 +67,9 @@ impl UserWallet {
 
-        bitcoin_client.faucet(p2tr_address.clone(), DEFAULT_FAUCET_AMOUNT)?;
-        sleep(Duration::from_secs(1)).await;
+        bitcoin_client
```
Move the faucet to a separate function

mhrynenko committed:
refactor: remove bitcoin cli node usage in test user_wallet

mhrynenko committed:
feat(gateway): split bitcoin client to support both node and external
api (esplora, mempool) broadcasting depending on config provided

mhrynenko committed:
fix(btc_indexer): update get rune addr to get both runes and regular
utxos, make confirmation blocks configurable, use mempool-aware
endpoints for some methods

mhrynenko committed:
fix(gateway): tx broadcast path

mhrynenko committed:
refactor(btc_indexer): use mempool related models, handle unwrap
gracefully

mhrynenko committed:
wip(tests): adjust spark test to be compatible with testnet4, decompose
bitcoin client into separate crate, add ability to use generated key in
tests

mhrynenko committed:
fix: network mappings, use proper issuer public key for token metadata

mhrynenko committed:
fix: adjust testnet4 test flow
